Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/EmulationActivity.java

```diff
@@ -748,71 +743,74 @@ public boolean dispatchKeyEvent(KeyEvent event)
 
   private void toggleControls()
   {
-    final SharedPreferences.Editor editor = mPreferences.edit();
-    boolean[] enabledButtons = new boolean[14];
     AlertDialog.Builder builder = new AlertDialog.Builder(this, R.style.DolphinDialogBase);
     builder.setTitle(R.string.emulation_toggle_controls);
     if (sIsGameCubeGame || mPreferences.getInt("wiiController", 3) == 0)
     {
-      for (int i = 0; i < enabledButtons.length; i++)
+      boolean[] gcEnabledButtons = new boolean[11];
+      String gcSettingBase = "MAIN_BUTTON_TOGGLE_GC_";
+
+      for (int i = 0; i < gcEnabledButtons.length; i++)
       {
-        enabledButtons[i] = mPreferences.getBoolean("buttonToggleGc" + i, true);
+        gcEnabledButtons[i] = BooleanSetting.valueOf(gcSettingBase + i).getBoolean(mSettings);
```

Comment:
I didn't know that this `valueOf` method existed, but that's pretty handy. An alternative way to accomplish this would be to use the `AdHocBooleanSetting` class, which would let us skip having to declare every possible controller setting in `BooleanSetting`... but I don't know, maybe declaring every possible controller setting is a good thing? If you want to keep declaring all controller settings, then keep using `valueOf`.

Reply:
You wouldn't need a map, but rather just passing `gcSettingBase + i` to the `AdHocBooleanSetting` constructor instead of to `BooleanSetting.valueOf`. But I'm okay with keeping this approach either way.

Reply:
Passing `gcSettingBase + i` as a key to `AdHocBooleanSetting` makes less informative keys when read from the INI, which is why I only see `AdHocBooleanSetting` being used with key maps.

Reply:
Oh, you're referring to how your current setup has integers in the enum names but the actual button names in the key names? Okay, I understand what you mean now.